api/client/gro_client.py

```diff
@@ -323,22 +323,36 @@ def find_data_series(self, **kwargs):
             search_results.append(
                 self.search('regions', kwargs['partner_region'])[:cfg.MAX_RESULT_COMBINATION_DEPTH])
             keys.append('partner_region_id')
-        all_data_series = []
+        # Rank by frequency and source, while preserving search ranking in
+        # permutations of item, metric, region, and partner region.
+        freq_ranking = []
+        ranking_groups = set()
+        count = 0
         for comb in itertools.product(*search_results):
             entities = dict(list(zip(keys, [entity['id'] for entity in comb])))
-            data_series_list = self.get_data_series(**entities)
-            self._logger.debug("Found {} distinct data series for {}".format(
-                len(data_series_list), entities))
-            # temporal coverage affects ranking so add time range if specified.
-            for data_series in data_series_list:
-                if kwargs.get('start_date'):
-                    data_series['start_date'] = kwargs['start_date']
-                if kwargs.get('end_date'):
-                    data_series['end_date'] = kwargs['end_date']
-            all_data_series += data_series_list
-        self._logger.warning("Found {} distinct data series total for {}".format(
-            len(all_data_series), kwargs))
-        for data_series in self.rank_series_by_source(all_data_series):
+            for data_series in self.get_data_series(**entities):
```

Comment:
i noticed that it will be very slow if any of the search_result actually expanded to a list of data series. for example, 
```
next(client.find_data_series(item="Rainfall",
                             metric="Precipitation Quantity",
                             region="United States of America"))
```
one of the  possible combination `({'id': 10081}, {'id': 2100031}, {'id': 1215})` will result in 50000 regions 
should we batch freq ranking for those expanded series (per each search result) ?
also otherwise, in next step,  `rank_series_by_source`  would be comparing expanded series with "at level"  search result?

Reply:
yes get_data_series() was originally  a generator (i.e. yields rather than return). however recent changes made it return a list intead of a generator...... I've put a limit of for now in cfg